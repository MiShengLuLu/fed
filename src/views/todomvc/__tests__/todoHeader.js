import { shallowMount } from '@vue/test-utils'
import TodoHeader from '../todoHeader.vue'

describe('todoHeader.vue', () => {
  let wrapper = null

  beforeEach(() => {
    wrapper = shallowMount(TodoHeader)
  })

  test('New todo', async () => {
    // 获取 input 输入框
    const input = wrapper.find('input[data-testid="new-todo"]')
    const text = 'NodeJs'
    await input.setValue(text)
    await input.trigger('keyup.enter')
    expect(wrapper.emitted()['new-todo']).toBeTruthy()
    expect(wrapper.emitted()['new-todo'][0][0]).toBe(text)
    expect(input.element.value).toBe('')
  })

  test('New todo no text', async () => {
    const input = wrapper.find('input[data-testid="new-todo"]')
    const text = ''
    await input.setValue(text)
    await input.trigger('keyup.enter')
    expect(wrapper.emitted()['new-todo']).toBeFalsy()
  })

  // 快照测试
  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
