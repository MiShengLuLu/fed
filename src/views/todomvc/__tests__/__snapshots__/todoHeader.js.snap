// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoHeader.vue snapshot 1`] = `
<header class="header">
  <h1>todos</h1><input data-testid="new-todo" class="new-todo" placeholder="What needs to be done?" autofocus="">
</header>
`;
